.github/workflows/tests.yml

```diff
@@ -10,33 +10,39 @@ on:
 
 jobs:
   tests:
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-24.04
     strategy:
       fail-fast: false
       matrix:
-        python-version: ["3.6", "3.7", "3.8", "3.9"]
-        django: [2.2.17, 3.0.11, 3.1.3, 3.2.0]
-        include:
-          - python-version: "3.10"
-            django: 3.2.0
-          - python-version: "3.8"
-            django: 4.0.0
+        python-version: ["3.8", "3.9", "3.10", "3.11", "3.12"]
+        django: ["3.2", "4.1", "4.2", "5.0", "main"]
+        exclude:
+          - python-version: "3.12"
+            django: "4.2" # Python 3.12 is supported since Django 4.2.8.
```

Comment:
No need to exclude it.  Even if you put `4.2.0` in the matrix, with `~=` it won't install `4.2.[0-7]`.

```suggestion
```

Reply:
Removed as suggested.